### PR TITLE
disable blobdb

### PR DESCRIPTION
**Pull request by phoenix-o**
BlobDB needs to be disabled (and the node needs to be resynced from scratch) before enabling effects pruning. 
The combination of BlobDB and pruning leads to high CPU utilization on the host, to the extent that the node stops making any progress. 
This happens because once a range delete is observed by a compaction job, RocksDB needs to deserialize the blob, remove the entry, serialize it again, and create a new blob multiple times

### CODE REVIEW

**Comment by sadhansood** [comment]
@phoenix-o would it be possible to share the underlying flamegraph, the screenshot is hard to parse (You could put it on a notion page maybe and add a link here). Thanks!

**Comment by phoenix-o** [comment]
@sadhansood: it's not a screenshot. Try to download it and open locally

**Comment by mwtian** [comment]
After disabling the option, can full compaction to the base layer avoid resync?

**Comment by mwtian** [comment]
Also, why do we need range delete on transactions and effects tables? IIUC these tables are keyed by digests?

**Comment by phoenix-o** [comment]
> After disabling the option, can full compaction to the base layer avoid resync?

hm, I can try, but I don't think so. We'd need to resync or restore from the snapshot

> why do we need range delete on transactions and effects tables

the same reason we do it for objects. Range deletes are stored in SST files in a separate section and don't have impact on reads until compaction job kicks in. But using point deletes here wouldn't prevent the problem. During compaction process rocksdb would still have to keep deserializing blobs, altering them, serializing and writing new ones

**Comment by sadhansood** [comment]
@phoenix-o just going through some rocksdb code, it seems like if we trigger a full compaction after disabling blob db - the values should move back to the sst file. It is worth doing that experiment and then if successful, we can likely leverage your background job to trigger compaction periodically to gradually do the transition.

**Comment by phoenix-o** [comment]
nice, will try that that a bit later and share results here

**Comment by phoenix-o** [comment]
quick update: (fewer) blob files still exist after compaction. Also effects compaction took 1.5 hours. It should be faster to restore from a snapshot

**Comment by sadhansood** [comment]
Thanks @phoenix-o. Could you also check how much blob data remains after running compaction (and if there is a setting to migrate away all blobs to sst). If it is only a small amount of data left as blobs, then does running tx+effects pruning result in the same kind of cpu overload? If not, maybe we can explore running periodic gradual background compaction to move blob data into sst.

The reason is if this works, we may not need coordinated manual snapshot restore (which a validator has to do mid epoch and incur downtime). What we could do instead is release a binary which disables blob db and enables your background compaction job to have old blob data migrated to sst files. And then in the following release, enable tx/effects pruning.

**Comment by phoenix-o** [comment]
3688 blob files. Overall blob size for perpetual db hasn't changed much, still north of 5gb 
Will check/investigate if such setting/mechanism exists

**Review by sadhansood — APPROVED** [review]

**Review by mwtian — APPROVED** [review]
Sounds good. I'm a bit surprised by the inefficiency. Maybe the min blob size needs to be even bigger.
